src/node/interfaces.cpp

```diff
@@ -989,6 +989,12 @@ class MinerImpl : public Mining
 
     std::unique_ptr<BlockTemplate> createNewBlock(const BlockCreateOptions& options) override
     {
+        {
+            // Ensure m_tip_block is set so consumers of BlockTemplate can rely on that.
+            WAIT_LOCK(notifications().m_tip_block_mutex, lock);
+            if (!notifications().TipBlock()) return nullptr;
```

Comment:
If it's not too difficult, can we reproduce this with a test?

Reply:
This depends on the node initialization and shutdown sequence. I'm not sure how to reproduce that in a test. And we might change that sequence in a way that this never happens in the first place.